common/cards/card-plugins/effects/chainmail-armor.js

```diff
@@ -0,0 +1,50 @@
+import {GameModel} from '../../../../server/models/game-model'
+import EffectCard from './_effect-card'
+
+/**
+ * @typedef {import('common/types/cards').CardPos} CardPos
+ */
+
+class ChainmailArmorEffectCard extends EffectCard {
+	constructor() {
+		super({
+			id: 'chainmail_armor',
+			name: 'Chainmail Armor',
+			rarity: 'common',
+			description: 'Prevents damage from all effect cards.',
+		})
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {CardPos} pos
+	 */
+	onAttach(game, instance, pos) {
+		const {otherPlayer} = pos
+
+		otherPlayer.hooks.onAttack[instance] = (attack, pickedSlots) => {
+			if (attack.target.index !== pos.rowIndex || attack.type === 'ailment')
+				return
+			if (attack.type === 'effect') {
+				attack.reduceDamage(attack.damage)
```

Comment:
you can just set the damage to 0 here, but hte important thing is to also lock the damage so it can never be added to

Reply:
I think it's better to use reduceDamage because of the lock, by setting it directly we are bypassing it.